File: calc_autovalores.m
% Author: Luciano de Oliveira Daniel
% https://sites.google.com/site/professorlucianodaniel

% Carrega matriz.
%
load('calc_autovalores_01.mat')
%
% Carrega autovalores python.
%
load('calc_autovalores_02.mat')
tic
eigM=eig(A);
toc
eigP=w';
calc_autovalores_plota(real(eigM),imag(eigM),real(eigP),imag(eigP))
%plot(real(eigM),imag(eigM),'r',real(eigP),imag(eigP), 'b')
[eigMs,I]=sort(imag(eigM));
[eigPs,J]=sort(imag(eigP));
% erro=abs((eigPs-eigMs)./eigMs)*100;
erro=abs((eigP(J)-eigM(I))./eigM(I))*100;
x=1:length(erro);
% plot(x,erro);
plota_erro(x,erro);

File: calc_autovalores.py
# Author: Luciano de Oliveira Daniel
# https://sites.google.com/site/professorlucianodaniel

from scipy.io import savemat
from numpy import random
from numpy import linalg
import time


def pause():
    input("Press the <ENTER> key to continue...")


dim = int(input('Dimension of the square random matrix:'))
A = random.rand(dim, dim)
print(A, '\n')
print()
savemat('calc_autovalores_01.mat', {'A': A})
t = time.time()
w = linalg.eigvals(A)
elapsed = time.time() - t
print(w, '\n')
print('EIG elapsed time in PYTHON (executable) is:', elapsed, 'seconds', '\n')
savemat('calc_autovalores_02.mat', {'w': w})
print('"I have a paper afloat which I hold to be great guns" (Maxwell, J.C.)', '\n')
pause()
